public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -113,6 +102,25 @@ export class PanelQueryRunner {
     );
   }
 
+  private getTransformationsStream = (withTransforms: boolean): MonoTypeOperatorFunction<PanelData> => {
+    return inputStream =>
+      inputStream.pipe(
+        mergeMap(data => {
+          if (!withTransforms) {
+            return of(data);
+          }
+
+          const transformations = this.dataConfigSource.getTransformations();
+
+          if (!transformations || transformations.length === 0) {
+            return of(data);
+          }
+
+          return transformDataFrame(transformations, data.series).pipe(map(series => ({ ...data, series })));
```

Comment:
why do we need this map?

Reply:
I think this should be the eqvivalent of:
```ts
processedData = {	
              ...processedData,	
              series: transformDataFrame(transformations, data.series),	
            };
```
I'm not 100% sure why the original code did that, but I agree that it looks kind of odd.

Reply:
ah, now I get it